piff/stats.py

```diff
@@ -0,0 +1,264 @@
+
+"""
+
+from __future__ import print_function
+import numpy as np
+
+from .psf import PSF
+
+def process_stats(config, logger):
+    """Parse the stats field of the config dict.
+
+    :param config:      The configuration dict.
+    :param logger:      A logger object for logging debug info.
+
+    :returns: an stats instance
+    """
+    import piff
+
+    if 'stats' not in config:
+        raise ValueError("config dict has no stats field")
+    config_stats = config['stats']
+
+    if 'type' not in config_stats:
+        raise ValueError("config['stats'] has no type field")
+
+    # Get the class to use for the stats
+    # Not sure if this is what we'll always want, but it would be simple if we can make it work.
+    stats_class = eval('piff.' + config_stats.pop('type'))
+
+    # Read any other kwargs in the stats field
+    kwargs = stats_class.parseKwargs(config_stats)
+
+    # Build stats object
+    stats = stats_class(**kwargs)
+
+    return stats
+
+class Statistics(PSF):
+    """The base class for getting the statistics of a set of stars.
+
+    It seems reasonable to use the PSF class as our base and build on top of that.
+    """
+
+    def stats(self, stars, logger=None, **kwargs):
+        """Perform your operation on the stars.
+
+        :param stars:       A list of StarData instances.
+        :param logger:      A logger object for logging debug info. [default: None]
+        :params kwargs:     Potential other parameters we might need to input. Images, coordinates, et cetera.
+
+        :returns:           Some kind of data vector.
+        """
+        raise NotImplemented("Derived classes must define the stats function")
+
+    def plot(self, fig=None, ax=None, logger=None, **kwargs):
+        """Make your plots.
+
+        :param fig:         A Matplotlib Figure object. [default: None]
+        :param ax:          A Matplotlib Axis object. [default: None]
+        :param logger:      A logger object for logging debug info. [default: None]
+        :params kwargs:     Potential other parameters we might need to input. Images, coordinates, et cetera.
+
+        :returns:           Some kind of data vector.
+        """
+        raise NotImplemented("Derived classes must define the plot function")
+
+class RhoStatistics(Statistics):
+    """Returns rho statistics using TreeCorr
+    """
+
+    def stats(self, stars, min_sep=1, max_sep=150, bin_size=0.1,
+              sky_coordinates=False, sep_units='rad',
+              ra_units='deg', dec_units='deg', logger=None):
+        """
+
+        :param stars:       A list of StarData instances.
+        :param logger:      A logger object for logging debug info. [default: None]
+        :param min_sep:     Minimum separation (in arcmin) for pairs
+        :param max_sep:     Maximum separation (in arcmin) for pairs
+        :param bin_size:    Logarithmic size of separation bins.
+        :param sky_coordinates: Are we in sky coordinates or otherwise? [default: False]
```

Comment:
This was done because most of the rho statistics plots you made were in arcmin, and I wasn't sure what the units of u,v are. The other part of this is that I think treecorr calculates the distances differently if u,v is on the sphere vs Euclidean (hence the ra/dec vs x/y).

By sky_coordinates always False, we then assume u,v is Euclidean. Is that right?